### DISCUSSION TITLE
Advice on new Meshtastic router build (solar powered + WiFi)

### DISCUSSION THREAD

**Discussion by vassilisv** [meshtastic/Discussions#22] [General]
I am just getting started with Meshtastic and would like to setup a repeater node at my home to increase coverage in my area. A couple of key requirements:
1. I would like the repeater to be able to connect to my home WiFi so that I can integrate it with my Home Assistant setup. 
2. I would also like to have the repeater be solar powered so that I can place it in higher and more favorable locations.

The combination of those two has proven somewhat difficult based on my research so far, as the battery and solar sizing will end up being too large to support the additional draw of the WiFi module (I am planning to use RAK Wisblock).

I was thinking that one option would be to utilize two nodes, one solar powered without WiFi setup as a router, then another node with WiFi in client mode that connects to my home network and is USB powered. Would that work and what would I be missing compared to a single node setup? Any other options I should be exploring? Thanks!

**Comment by TerryFrench** [comment]
Looks like that would work. If this is your home, would POE be an option to power a more power-hungry node?

**Comment by madeofstown** [comment]
If you want to run a solar node with WiFi abilities something with an ESP-32 chip is going to be your best bet, preferably with a way to attach an external antenna.

**Reply by h0lad** [reply]
ESP32+Wifi doesn't make really fun with small solar cell. With that mode activated it will suck power like crazy.

**Reply by madeofstown** [reply]
> ESP32+Wifi doesn't make really fun with small solar cell. With that mode activated it will suck power like crazy.

It's the best optiion available if OP wants WiFi and Solar. The only other option is Linux native which is more power hungry. nRF MCUs don't support WiFi

**Reply by barryhunter** [reply]
I believe Pi Pico (RP2040) MCU devices are reasonably power efficient, better than ESP32 anyway. 
And should support basic wifi, using a Pi PicoW (enough for MQTT I believe). (although wifi in itself, is going to contribute a lot to the power usage) 

A PicoW is a bit more 'homebrew' as need to use an external Lora module and connect it to a PicoW. Dont think there are any ready made RP2024+Wifi+Lora chipset boards. (there are RP2024+Lora, and there are RP2040+Wifi, but not all 3 in one)

**Reply by madeofstown** [reply]
> I believe Pi Pico (RP2040) MCU devices are reasonably power efficient, better than ESP32 anyway. And should support basic wifi, using a Pi PicoW (enough for MQTT I believe). (although wifi in itself, is going to contribute a lot to the power usage)
> 
> A PicoW is a bit more 'homebrew' as need to use an external Lora module and connect it to a PicoW. Dont think there are any ready made RP2024+Wifi+Lora chipset boards. (there are RP2024+Lora, and there are RP2040+Wifi, but not all 3 in one)

I always forget about picoW even though that was the first node I built. Thanks for the reminder. OP, Waveshare makes a nice LoRa hat(?) for the pico that has a built in battery charger circuit.

**Comment by barryhunter** [comment]
The only 'drawback' (other than cost etc) of a two node setup I can see, is your 'client' node will be an extra hop away from other users. 

Eg the standard HopLimit is 3, so your 'Router' node would see users within 3 hops.. But your client node, would only see nodes within 2 hops of the router. (assuming that the client node, wouldnt have much direct contact with remote nodes) 

YEs, can raise the hoplimit on your client for outgoing, but doesnt work in reverse!

Being able to connect to the router, directly with wifi wouldnt have direct benefit. Wouldnt be able to update firmware like that for example. 

Having MQTT gateway on your indoor client, wouldnt really be any different to it being on the router (other than the hoplimit range talked about)

**Comment by chixxi** [comment]
I tried the same thing, but I was not successful since the Liliygo T3S3 was pulling 200mA when connected to Wifi. With the rather cloudy and foggy days here in Switzerland that was to much for the 5 Watt Solar panel I was using, sooner or later it would go empty, no matter how big the battery was.

I am now using a Wisblock with the RAK4631 Core, that is using 22mW. Exactly as @barryhunter mentioned that gives me the disadvantage of loosing a hop since I still run a plugged in T3S3 indoors which is connected to Wifi so that I can use Node-Red.
